File: Java-CodeServer/src/main/resources/js/prompto/runtime/Variable.js
var BaseType = require("../type/BaseType").BaseType;

function Variable (name, type) {
    if(!(type instanceof BaseType))
        throw new Exception();
    this.name = name;
	this.type = type;
	return this;
}

Variable.prototype.toString = function() {
	return this.name;
}

Variable.prototype.getType = function(context) {
    return this.type;
};

exports.Variable = Variable;

File: Core/src/main/java/prompto/runtime/Variable.java
package prompto.runtime;

import prompto.grammar.INamedInstance;
import prompto.grammar.Identifier;
import prompto.type.IType;

public class Variable implements INamedInstance {

	Identifier name;
	IType type;
	
	public Variable(Identifier name, IType type) {
		this.name = name;
		this.type = type;
	}

	@Override
	public String toString() {
		return name.toString();
	}
	
	@Override
	public Identifier getId() {
		return name;
	}

	@Override
	public IType getType(Context context) {
		return type;
	}
	
}
